tensorizer/serialization.py

```diff
@@ -432,11 +432,22 @@ def _read_shapes(obj, num_elems) -> List[int]:
 
     @property
     def total_bytes_read(self) -> int:
+        if hasattr(self._file, "bytes_read"):
+            return self._file.bytes_read
         if self._file.closed:
             return self.total_tensor_bytes
         else:
             return self._file.tell()
 
+    # If our _file object has 'response_headers' attribute, we can use it
+    # to determine if we were cached or not.
+    @property
+    def cache_status(self) -> Union[bool, str]:
+        if hasattr(self._file, "response_headers"):
+            return self._file.response_headers.get("x-cache-status", False)
+        else:
+            return False
```

Comment:
This would be better as an `Optional[str]`
```suggestion
    @property
    def cache_status(self) -> Optional[str]:
        if hasattr(self._file, "response_headers"):
            return self._file.response_headers.get("x-cache-status", None)
        else:
            return None
```